### PR TITLE
TPC QC: adds IDCvsSACZero Task

### PR DESCRIPTION
**Pull request by sbhawani**

### CODE REVIEW

File: Detectors/TPC/qc/include/TPCQC/IDCsVsSACs.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#ifndef AliceO2_TPC_IDCSVSSACS_H
+#define AliceO2_TPC_IDCSVSSACS_H
+
+// root includes
+#include "TCanvas.h"
+
+// o2 includes
+#include "TPCCalibration/IDCCCDBHelper.h"
+#include "TPCCalibration/SACCCDBHelper.h"
+#include "TPCCalibration/IDCContainer.h"
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/IDCsVsSACs.h]:
```suggestion
```

File: Detectors/TPC/qc/include/TPCQC/IDCsVsSACs.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#include "DataFormatsTPC/Defs.h"
+
+class IDCCCDBHelper;
+class SACCCDBHelper;
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/IDCsVsSACs.h]:
```suggestion
class TCanvas;

namespace o2::tpc {
template <class>
class IDCCCDBHelper;

template <class>
class SACCCDBHelper;
}

```

File: Detectors/TPC/qc/include/TPCQC/IDCsVsSACs.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+///
+/// @file   IDCsVsSACs.h
+/// @author Bhawani Singh
+///
+
+#ifndef AliceO2_TPC_IDCSVSSACS_H
+#define AliceO2_TPC_IDCSVSSACS_H
+
+// root includes
+#include "TCanvas.h"
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/IDCsVsSACs.h]:
```suggestion
#include <string>
```

File: Detectors/TPC/qc/src/IDCsVsSACs.cxx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+// or submit itself to any jurisdiction.
+
+#include "TPCQC/IDCsVsSACs.h"
+#include "TPCCalibration/IDCCCDBHelper.h"
+#include "TPCCalibration/IDCDrawHelper.h"
+#include "TPCCalibration/SACDrawHelper.h"
+#include "TPCCalibration/SACFactorization.h"
+#include "TPCCalibration/IDCGroupHelperSector.h"
+#include "TPCCalibration/IDCContainer.h"
+#include "TPCCalibration/IDCContainer.h"
```
Comment (wiechula) [inline Detectors/TPC/qc/src/IDCsVsSACs.cxx]:
```suggestion
#include "TPCCalibration/IDCContainer.h"
```

File: Detectors/TPC/qc/src/IDCsVsSACs.cxx [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "TPCQC/IDCsVsSACs.h"
+#include "TPCCalibration/IDCCCDBHelper.h"
```
Comment (wiechula) [inline Detectors/TPC/qc/src/IDCsVsSACs.cxx:21]:
```suggestion
#include "TPCCalibration/IDCCCDBHelper.h"
#include "TPCCalibration/SACCCDBHelper.h"
```

File: Detectors/TPC/qc/src/IDCsVsSACs.cxx [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+#include "TPCBase/Mapper.h"
+#include "CommonUtils/TreeStreamRedirector.h"
+#include "TPCBase/Painter.h"
+
+#include "TStyle.h"
+#include "TCanvas.h"
+#include "TH2Poly.h"
+#include "TPCCalibration/IDCFactorization.h"
+#include <map>
+#include "TProfile.h"
```
Comment (wiechula) [inline Detectors/TPC/qc/src/IDCsVsSACs.cxx]: Please remove unneded headers

File: Detectors/TPC/qc/src/IDCsVsSACs.cxx [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+  };
+
+  SACDrawHelper::SACDraw drawFun;
+  drawFun.mSACFunc = SACFunc;
+  const std::string zAxisTitle = SACDrawHelper::getZAxisTitle(o2::tpc::SACType::IDCZero);
+
+  auto hSac0SideA = SACDrawHelper::drawSide(drawFun, o2::tpc::Side::A, zAxisTitle);
+  auto hSac0SideC = SACDrawHelper::drawSide(drawFun, o2::tpc::Side::C, zAxisTitle);
+
+  std::string SACname = "SAC0";
```
Comment (wiechula) [inline Detectors/TPC/qc/src/IDCsVsSACs.cxx]:
```suggestion
  const std::string SACname = "SAC0";
```

File: Detectors/TPC/qc/src/IDCsVsSACs.cxx [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+  c->Clear();
+  c->Divide(2, 3);
+
+  c->cd(1);
+  hAside2D->Draw("colz");
+  hAside2D->SetStats(0);
+  hAside2D->SetTitleOffset(1.05, "XY");
+  hAside2D->SetTitleSize(0.05, "XY");
+  // drawSectorsXY(Side::A);
```
Comment (wiechula) [inline Detectors/TPC/qc/src/IDCsVsSACs.cxx]: Why did you remove it?

**Review by wiechula — CHANGES_REQUESTED** [review]
Just minor things. In the QC meeting we also discussed to add ratio histograms. Please check with @bulukutlu

File: Detectors/TPC/qc/include/TPCQC/IDCsVsSACs.h [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+class TCanvas;
+namespace o2::tpc
+{
+template <class>
+class IDCCCDBHelper;
+template <class>
+class SACCCDBHelper;
+} // namespace o2::tpc
+// namespace o2::tpc
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/IDCsVsSACs.h:34]:
```suggestion

```

**Review by wiechula — APPROVED** [review]

**Comment by sbhawani** [comment]
Hi, can this be merged? thanks

**Pull request merged**